src/components/Admin/PromptCard.vue

```diff
@@ -239,6 +239,11 @@ async function onSubmit() {
     $q.notify({ type: 'negative', message: 'Choose another month for this prompt.' })
     return
   }
+  const hasPrompt = await promptStore.hasPrompt(prompt.date, prompt.title, prompt.slug)
+
+  if (hasPrompt) {
+    return
+  }
```

Comment:
This hasPrompt is related to **prompt url slug edge case** #479  issue. We locally find if there is any prompt with a similar title or 'month' title. If there is, then we show an error notification back. We can only add prompts from the Admin panel and in the Admin panel we fetch all the prompts so we can locally compare the title, however, in this current implementation we can add prompts from anywhere so we might not have all the prompt data locally so even if we compare there might be a case where we will add the same title, so to prevent that I have implemented this.

Reply:
Ok I got your idea. But there are still some issue with this approach. 
1. When creating a prompt from any page other than Search page (so promptsore.getPrompts is undefined), after submitting the prompt we get an error but the prompt is still created.
2. There is still a way to reduce the amount of fetches. We could add an if statement, and call promptstore.hasPrompt only if promptstore.getPrompts is undefined. So if there are "locally" stored prompts there is probably no need for now to call hasPrompts again. 
`  if (!promptStore.getPrompts.length) {
    const hasPrompt = await promptStore.hasPrompt(prompt.date, prompt.title, prompt.slug)
    if (hasPrompt) {
      console.log(promptStore.getPrompts)
      console.log(hasPrompt)
      return
    }
  }`

Though your approach can still be probably usefull in case when two prompts are created simultaneously by two different users (if that could happen), and one doesn't have the updated prompts list. 

![Screenshot 2024-08-30 at 14 19 51](https://github.com/user-attachments/assets/3d427516-b618-47ac-b9ac-18959bcee63a)
![Screenshot 2024-08-30 at 14 20 04](https://github.com/user-attachments/assets/4c4f320f-d483-41cc-9138-d1d52e0fab68)